### ISSUE TITLE
[MNG-6024] maven-antrun-plugin:instrument failing NoClassDefFoundError: org/slf4j/LoggerFactory

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7849]
**[S V Mohana Rao](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mohanaraosv)** opened **[MNG-6024](https://issues.apache.org/jira/browse/MNG-6024?redirect=false)** and commented

Related issues : MNG-5783, MNG-5817. I was using maven-antrun-plugin added dependency cobertura which requires slf4j-api dependent jar. Even i tried using 3.4.0-SNAPSHOT version but problem persists. As i was observed it's been excluded from the child dependent artifacts of cobertura cause it's part maven core. But it's not referring from maven core which hasn't been added class path.

---

**Affects:** 3.3.1, 3.3.3, 3.3.9

**Attachments:**
- [maven-core-slf4j-issue.zip](https://issues.apache.org/jira/secure/attachment/12824141/maven-core-slf4j-issue.zip) (_41.54 kB_)
- [mvn-coreslf4j-issue.txt](https://issues.apache.org/jira/secure/attachment/12804395/mvn-coreslf4j-issue.txt) (_7.59 kB_)

**Issue Links:**
- [MNG-6031](https://issues.apache.org/jira/browse/MNG-6031) Moving from Maven 3.2.5 or older to Maven 3.3.1 or newer breaks maven-antrun-plugin
 (_**"is duplicated by"**_)
- [MNG-5783](https://issues.apache.org/jira/browse/MNG-5783) cobertura-maven-plugin:instrument failing NoClassDefFoundError: org/slf4j/LoggerFactory

- [MNG-5817](https://issues.apache.org/jira/browse/MNG-5817) Maven 3.3, gradle-maven-plugin fails missing class org/slf4j/impl/NewMojoLogger

0 votes, 8 watchers

**Comment by jira-importer** [comment]
**[S V Mohana Rao](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mohanaraosv)** commented

Added maven console output.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Please provide a fully working example.

**Comment by jira-importer** [comment]
**[S V Mohana Rao](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mohanaraosv)** commented

Okay, I will create a sample maven project which simulates the above scenario.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Am I still waiting for your example.

**Comment by jira-importer** [comment]
**[S V Mohana Rao](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mohanaraosv)** commented

Please run the following command to simulate error.

mvn clean install
mvn verify -Dcobertura=true -Pcobertura

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I can confirm this.

Dependency tree before:

```
[DEBUG] org.apache.maven.plugins:maven-antrun-plugin:jar:1.8:
[DEBUG]    net.sourceforge.cobertura:cobertura:jar:2.1.1:runtime
[DEBUG]       org.ow2.asm:asm:jar:5.0.1:runtime
[DEBUG]       org.ow2.asm:asm-tree:jar:5.0.1:runtime
[DEBUG]       org.ow2.asm:asm-commons:jar:5.0.1:runtime
[DEBUG]       org.ow2.asm:asm-util:jar:5.0.1:runtime
[DEBUG]       org.ow2.asm:asm-analysis:jar:5.0.1:runtime
[DEBUG]       oro:oro:jar:2.0.8:runtime
[DEBUG]       jaxen:jaxen:jar:1.1.4:runtime
[DEBUG]       org.apache.ant:ant-junit:jar:1.8.3:runtime
[DEBUG]       org.mortbay.jetty:servlet-api-2.5:jar:6.1.14:runtime
[DEBUG]       org.mortbay.jetty:jetty:jar:6.1.14:runtime
[DEBUG]       org.mortbay.jetty:jetty-util:jar:6.1.14:runtime
[DEBUG]       com.sun:tools:jar:0:system
[DEBUG]       org.slf4j:slf4j-api:jar:1.7.5:runtime
[DEBUG]       ch.qos.logback:logback-classic:jar:1.0.13:runtime
[DEBUG]          ch.qos.logback:logback-core:jar:1.0.13:runtime
[DEBUG]       junit:junit:jar:4.11:compile
[DEBUG]          org.hamcrest:hamcrest-core:jar:1.3:compile
[DEBUG]       org.apache.commons:commons-lang3:jar:3.3.2:runtime
[DEBUG]    org.apache.maven:maven-plugin-api:jar:2.2.1:compile
[DEBUG]    org.apache.maven:maven-project:jar:2.2.1:compile
[DEBUG]       org.apache.maven:maven-settings:jar:2.2.1:compile
[DEBUG]       org.apache.maven:maven-profile:jar:2.2.1:compile
[DEBUG]       org.apache.maven:maven-model:jar:2.2.1:compile
[DEBUG]       org.apache.maven:maven-artifact-manager:jar:2.2.1:compile
[DEBUG]          org.apache.maven:maven-repository-metadata:jar:2.2.1:compile
[DEBUG]          backport-util-concurrent:backport-util-concurrent:jar:3.1:compile
[DEBUG]       org.apache.maven:maven-plugin-registry:jar:2.2.1:compile
[DEBUG]       org.codehaus.plexus:plexus-interpolation:jar:1.11:compile
[DEBUG]       org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9-stable-1:compile
[DEBUG]          classworlds:classworlds:jar:1.1-alpha-2:compile
[DEBUG]    org.apache.maven:maven-artifact:jar:2.2.1:compile
[DEBUG]    org.codehaus.plexus:plexus-utils:jar:3.0.20:compile
[DEBUG]    org.apache.ant:ant:jar:1.9.4:compile
[DEBUG]       org.apache.ant:ant-launcher:jar:1.9.4:compile
```

and after population

```
[DEBUG] Importing foreign packages into class realm plugin>org.apache.maven.plugins:maven-antrun-plugin:1.8
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>org.apache.maven.plugins:maven-antrun-plugin:1.8
[DEBUG]   Included: org.apache.maven.plugins:maven-antrun-plugin:jar:1.8
[DEBUG]   Included: net.sourceforge.cobertura:cobertura:jar:2.1.1
[DEBUG]   Included: org.ow2.asm:asm:jar:5.0.1
[DEBUG]   Included: org.ow2.asm:asm-tree:jar:5.0.1
[DEBUG]   Included: org.ow2.asm:asm-commons:jar:5.0.1
[DEBUG]   Included: org.ow2.asm:asm-util:jar:5.0.1
[DEBUG]   Included: org.ow2.asm:asm-analysis:jar:5.0.1
[DEBUG]   Included: oro:oro:jar:2.0.8
[DEBUG]   Included: jaxen:jaxen:jar:1.1.4
[DEBUG]   Included: org.apache.ant:ant-junit:jar:1.8.3
[DEBUG]   Included: org.mortbay.jetty:servlet-api-2.5:jar:6.1.14
[DEBUG]   Included: org.mortbay.jetty:jetty:jar:6.1.14
[DEBUG]   Included: org.mortbay.jetty:jetty-util:jar:6.1.14
[DEBUG]   Included: com.sun:tools:jar:0
[DEBUG]   Included: ch.qos.logback:logback-classic:jar:1.0.13
[DEBUG]   Included: ch.qos.logback:logback-core:jar:1.0.13
[DEBUG]   Included: junit:junit:jar:4.11
[DEBUG]   Included: org.hamcrest:hamcrest-core:jar:1.3
[DEBUG]   Included: org.apache.commons:commons-lang3:jar:3.3.2
[DEBUG]   Included: backport-util-concurrent:backport-util-concurrent:jar:3.1
[DEBUG]   Included: org.codehaus.plexus:plexus-interpolation:jar:1.11
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.0.20
[DEBUG]   Included: org.apache.ant:ant:jar:1.9.4
[DEBUG]   Included: org.apache.ant:ant-launcher:jar:1.9.4
[DEBUG] Configuring mojo org.apache.maven.plugins:maven-antrun-plugin:1.8:run from plugin realm ClassRealm[plugin>org.apache.maven.plugins:maven-antrun-plugin:1.8, parent: sun.misc.Launcher$AppClassLoader@5c647e05]
```

**Comment by jira-importer** [comment]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** commented

ok, lets close issue as it would be hard to do something with not supported cobertura.
